packages/provider/src/language-model/v2/language-model-v2-prompt.ts

```diff
@@ -0,0 +1,149 @@
+import { JSONValue } from '../../json-value/json-value';
+
+/**
+A prompt is a list of messages.
+
+Note: Not all models and prompt formats support multi-modal inputs and
+tool calls. The validation happens at runtime.
+
+Note: This is not a user-facing prompt. The AI SDK methods will map the
+user-facing prompt types such as chat or instruction prompts to this format.
+ */
+export type LanguageModelV2Prompt = Array<LanguageModelV2Message>;
+
+export type LanguageModelV2Message =
+  | {
+      role: 'system';
+      content: string;
+    }
+  | {
+      role: 'user';
+      content: Array<
+        | LanguageModelV2TextPart
+        | LanguageModelV2DataPart
+        | LanguageModelV2ToolResultPart
+      >;
+
+      /**
+       * Additional provider-specific metadata. They are passed through
+       * to the provider from the AI SDK and enable provider-specific
+       * functionality that needs to be applied per message,
+       * e.g. the OpenAI name parameter.
+       */
+      providerMetadata: Record<string, JSONValue> | undefined;
```

Comment:
I am probably missing something but couldn't you just add this field in a backwards compatible way to the LanguageModelV1 types since the property is optional?

Reply:
Good point. My preference is to move to a cleaner language model, but considering the need to get prompt caching ready sooner rather than later I'll see if that's an option.